src/platform/plugins/shared/discover/public/application/main/state_management/redux/runtime_state.tsx

```diff
@@ -49,7 +48,7 @@ type ReactiveRuntimeState<TState, TNullable extends keyof TState = never> = {
 export type ReactiveTabRuntimeState = ReactiveRuntimeState<TabRuntimeState, 'currentDataView'>;
 
 export type RuntimeStateManager = ReactiveRuntimeState<DiscoverRuntimeState> & {
-  tabs: { byId: Record<string, ReactiveTabRuntimeState | undefined> };
+  tabs: { byId: Record<string, ReactiveTabRuntimeState> };
```

Comment:
I reverted the type change to the runtime tab state map and associated changes because I don't think we should be handling it that way. It's intended to stay in sync with the Redux store and always have an available runtime tab state for any ID in `tabs.allIds`. If that ever isn't the case then we have a state management bug, and I think it should be treated as an actual error scenario. Maybe there's a good reason to change this, but I'd like to leave it as is for now and discuss it more before making any changes.